### PR TITLE
[FLINK-29548][flink-test-utils] Remove deprecated class files of the 'flink-test-utils' module.

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

Remove deprecated class files of the 'flink-test-utils' module.

## Brief change log

*Only code-cleanup change.*
Delete 
- flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/MiniClusterResource.java

- flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/MiniClusterResourceConfiguration.java

## Verifying this change

This change is a  code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by RocMarshal** [comment]
@gaoyunhaii Could you help to check it ? thx~

File: flink-table/flink-table-api-scala/src/main/scala/org/apache/flink/table/api/ImplicitExpressionConversions.scala [hunk 1]
```diff
@@ -496,16 +496,16 @@ trait ImplicitExpressionConversions {
   }
 
   /**
-   * Converts the date time string with format 'yyyy-MM-dd HH:mm:ss' under the 'UTC+0' time zone
-   * to a timestamp value of [[DataTypes.TIMESTAMP]].
+   * Converts the date time string with format 'yyyy-MM-dd HH:mm:ss' under the 'UTC+0' time zone to
+   * a timestamp value of [[DataTypes.TIMESTAMP]].
```
Comment (zentol) [inline flink-table/flink-table-api-scala/src/main/scala/org/apache/flink/table/api/ImplicitExpressionConversions.scala:500]: What is this change about?

Comment (RocMarshal) [reply flink-table/flink-table-api-scala/src/main/scala/org/apache/flink/table/api/ImplicitExpressionConversions.scala:500]: https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=41747&view=logs&j=52b61abe-a3cc-5bde-cc35-1bbe89bb7df5&t=54421a62-0c80-5aad-3319-094ff69180bb 

The code style check failed.

Comment (zentol) [reply flink-table/flink-table-api-scala/src/main/scala/org/apache/flink/table/api/ImplicitExpressionConversions.scala:500]: CI is passing though?

Comment (zentol) [reply flink-table/flink-table-api-scala/src/main/scala/org/apache/flink/table/api/ImplicitExpressionConversions.scala:500]: I also don't get it locally...

Comment (zentol) [reply flink-table/flink-table-api-scala/src/main/scala/org/apache/flink/table/api/ImplicitExpressionConversions.scala:500]: This was already fixed in 82e279ee.

**Comment by zentol** [comment]
Please rebase the PR and then we can merge it I think.

**Pull request merged**